dashboard/app/controllers/ai_diff_controller.rb

```diff
@@ -0,0 +1,66 @@
+class AiDiffController < ApplicationController
+  include AiDiffBedrockHelper
+  # authorize_resource class: false
+  before_action :require_levelbuilder_mode_or_test_env, except: [:chat_completion]
+  load_and_authorize_resource except: [:chat_completion]
+
+  # params are
+  # inputText:
+  # lessonId:
+  # unitDisplayName:
+  # sessionId:
+  # POST /ai_diff/chat_completion
+  def chat_completion
+    return render status: :forbidden, json: {} unless current_user&.teacher?
+    is_ai_diff_experiment_enabled = current_user && SingleUserExperiment.enabled?(user: current_user, experiment_name: 'ai-differentiation')
+    return render status: :forbidden, json: {} unless is_ai_diff_experiment_enabled
+    unless has_required_params?
+      return render status: :bad_request, json: {}
+    end
+
+    @session_id = params[:sessionId].presence ? params[:sessionId] : nil
```

Comment:
Should we be storing these session IDs somewhere associated with a user so that
a) We can check to make sure the current user actually initiated this session rather than letting users hijack other users' sessions if they somehow discover the ID, and
b) A user can resume a chat where they left off if they move to a different device or otherwise lose their front-end context?